File: programs/program_03.c
#include <stdio.h>

int main(void) {
    int n, top;
    printf("Input max array size : ");
    scanf("%d", &n);
    int a[n];
    printf("Input number of array elements : ");
    scanf("%d", &top);
    printf("Input array elements : ");
    for (int i = 0; i < top; i++)
        scanf("%d", &a[i]);
    printf("Array elements : ");
    for (int i = 0; i < top; i++)
        printf("%d ", a[i]);
    int mid = top / 2;
    for (int i = top; i > mid; i--)
        a[i] = a[i - 1];
    printf("\nInput the element to be inserted : ");
    scanf("%d", &a[mid]);
    top++;
    printf("Array elements after insertion : ");
    for (int i = 0; i < top; i++)
        printf("%d ", a[i]);
    printf("\n");
    return 0;
}

File: programs/program_03.rs
fn main() {
    let mut arr = [0; 100];
    println!("Input array size :");
    let mut size = String::new();
    std::io::stdin().read_line(&mut size).unwrap();
    let size: usize = size.trim().parse().unwrap();
    println!("Input array elements :");
    for x in 0..size {
        let mut input = String::new();
        std::io::stdin().read_line(&mut input).unwrap();
        let input: i32 = input.trim().parse().unwrap();
        arr[x] = input;
    }
    let mut i = size;
    let mid = size / 2;
    while i > mid {
        arr[i] = arr[i - 1];
        i -= 1;
    }
    println!("Input element to be inserted :");
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();
    arr[mid] = input.trim().parse().unwrap();
    println!("Array after insertion :");
    for x in 0..(size + 1) {
        print!("{} ", arr[x]);
    }
    println!("");
}
